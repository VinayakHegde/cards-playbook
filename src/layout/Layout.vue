<script src="./index.js"></script>
<style lang="scss" src="./style.scss"></style>
<template>
  <div id="cards-playbook">
    <Header v-bind:text="text" />
    <router-view />
  </div>
</template>
