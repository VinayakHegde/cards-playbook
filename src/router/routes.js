export default [
  {
    path: "/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/",
    redirect: {
      name: "Home"
    }
  },
  {
    path: "/deck/new",
    name: "Home",
    meta: {
      header: "CARDS"
    },
    component: () =>
      import(/* webpackChunkName: "home" */ "../views/home/Home.vue")
  },
  {
    path: "/deck/:hash",
    name: "Pack",
    meta: {
      header: "Ordered Pile"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/pack/Pack.vue"),
    props: true
  },
  {
    path: "/error",
    name: "Error",
    meta: {
      header: "404 | Page not found"
    },
    component: () =>
      import(/* webpackChunkName: "Pack" */ "../views/error/Error.vue"),
    props: true
  },
  {
    path: "*",
    redirect: {
      name: "Error"
    }
  }
];
