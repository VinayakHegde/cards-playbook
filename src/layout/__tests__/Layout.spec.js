import VueRouter from "vue-router";
import { mount, createLocalVue  } from "@vue/test-utils";
import Layout from "../Layout.vue";
import routes from "../../router/routes";

const localVue = createLocalVue();
localVue.use(VueRouter);
describe("Layout.vue", () => {
  it("match the snapshot", () => {
    const router = new VueRouter({ routes });
    const wrapper = mount(Layout, {
      localVue,
      router
    });
    expect(wrapper).toMatchSnapshot();
  });
});
