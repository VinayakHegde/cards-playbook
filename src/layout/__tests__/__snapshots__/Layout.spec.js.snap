// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout.vue match the snapshot 1`] = `
<div id="cards-playbook">
  <header></header>
  <!---->
</div>
`;
